apps/recovery-portal/src/main/webapp/password-recovery-tenant-request.jsp

```diff
@@ -0,0 +1,158 @@
+                    <%
+                        if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                    %>
+                    <input id="tenant-domain" type="text" name="tenantDomain"
+                                class="form-control ">
+                    <%
+                        }
+                    %>
+                        <%
+                            String callback = Encode.forHtmlAttribute
+                                    (request.getParameter("callback"));
+                            if (callback != null) {
+                        %>
+                        <div>
+                            <input type="hidden" name="callback" value="<%=callback %>"/>
+                        </div>
+                        <%
+                            }
+                        %>
+                        <div class="ui divider hidden"></div>
+
+                        <div class="align-right buttons">
+                            <a href="javascript:goBack()" class="ui button link-button">
+                                <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "Cancel")%>
+                            </a>
+                            <button id="recoverSubmit"
+                                    class="ui primary large button"
+                                    type="submit">
+                                <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
+                                "Proceed.password.recovery")%>
+                            </button>
+                        </div>
+                </form>
+            </div>
+        </div>
+    </div>
+</main>
+
+
+    <!-- product-footer -->
+    <%
+        File productFooterFile = new File(getServletContext().getRealPath("extensions/product-footer.jsp"));
+        if (productFooterFile.exists()) {
+    %>
+        <jsp:include page="extensions/product-footer.jsp"/>
+    <% } else { %>
+        <jsp:include page="includes/product-footer.jsp"/>
+    <% } %>
+
+    <!-- footer -->
+    <%
+        File footerFile = new File(getServletContext().getRealPath("extensions/footer.jsp"));
+        if (footerFile.exists()) {
+    %>
+        <jsp:include page="extensions/footer.jsp"/>
+    <% } else { %>
+        <jsp:include page="includes/footer.jsp"/>
+    <% } %>
+
+    <script type="text/javascript">
+        function goBack() {
+            window.history.back();
+        }
+
+        $(document).ready(function () {
+            $("#tenantBasedRecovery").submit(function (e) {
+                var errorMessage = $("#error-msg");
+                errorMessage.hide();
+                var tenantDomain = $("#tenant-domain").val();
+                var isTenantQualifiedUrlsEnabled = '<%= IdentityTenantUtil.isTenantQualifiedUrlsEnabled() %>';
+
+                if (isTenantQualifiedUrlsEnabled == 'false' && tenantDomain == '') {
+                    errorMessage.text("Please enter your tenant domain.");
```

Comment:
This error message should be localized as well right?

Reply:
Addressed in https://github.com/wso2/identity-apps/pull/1245/commits/a2b9c00bf6d0fb5c25efcf61feff8db6e2b45cff